taskvine/src/worker/vine_cache.h

```diff
@@ -45,8 +50,9 @@ void vine_cache_scan( struct vine_cache *c, struct link *manager );
 char *vine_cache_full_path( struct vine_cache *c, const char *cachename );
 
 int vine_cache_addfile( struct vine_cache *c, int64_t size, int mode, const char *cachename );
-int vine_cache_queue_transfer( struct vine_cache *c, const char *source, const char *cachename, int64_t size, int mode );
+int vine_cache_queue_transfer( struct vine_cache *c, const char *source, const char *cachename, int64_t size, int mode, int flags );
```

Comment:
Typically, `flags` implies some bit values or-ed together.  Is that your intention here?  If so, then use the enumeration in the type, and call things consistently `VINE_CACHE_FLAGS_X` and `vine_cache_flags_t` with assigned values.  (And then check for `flags&value` instead of `flags==value`.)

If this is meant to be restricted to a single integer value indicating when to act, then call the parameter `vine_cache_when_t when` and make the values `VINE_CACHE_WHEN_X` etc so that it is unavoidably clear what goes where.